src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs

```diff
@@ -979,40 +979,26 @@ private bool SaveAsOnnxCore(OnnxContext ctx, string[] outputs, string featureCol
         {
             Host.CheckValue(ctx, nameof(ctx));
 
-            string predictedLabelInt64 = null;
-            string predictedLabelUint32 = null;
-            // REVIEW: What is the right way to get the name of the predicted column?
-            for (int i = 0; i < outputs.Length; i++)
-            {
-                if (outputs[i] != DefaultColumnNames.PredictedLabel)
-                    continue;
-                predictedLabelUint32 = DefaultColumnNames.PredictedLabel;
-                predictedLabelInt64 = ctx.AddIntermediateVariable(NumberDataViewType.Int64, "PredictedLabelInt64", true);
-                outputs[i] = predictedLabelInt64;
-                break;
-            }
-
-            Host.CheckValue(predictedLabelInt64, nameof(predictedLabelInt64));
+            string labels = ctx.AddIntermediateVariable(NumberDataViewType.Int64, "Labels", true);
 
             string opType = "LinearClassifier";
-            var node = ctx.CreateNode(opType, new[] { featureColumn }, outputs, ctx.GetNodeName(opType));
+            var node = ctx.CreateNode(opType, new[] { featureColumn }, new[] { labels, outputs[1] }, ctx.GetNodeName(opType));
```

Comment:
Is the assumption here that the outputs array contain strings in a fixed order? If so, can you please add an assert about what you are expecting in outputs[1] ?(Maybe upon function entry along with the checks)

Reply:
yup, that's the assumption that many other trainers have